### DISCUSSION TITLE
Migrating issues - mismatched PostgreSQL versions

### DISCUSSION THREAD

**Discussion by chamik** [plausible/analytics#2792] [Self-Hosted Support] [answered]
Hi, I have a nicely working instance of plausible and I want to migrate it to a new server.

I've made a backup of docker volumes (using [this guide](https://nicholas.cloud/blog/backing-up-and-restoring-a-self-hosted-plausible-instance/)) and successfully mounted them to the new instance.

However, when I try to run the new instance I get the following errors:

```
plausible-hosting-plausible_db-1 exited with code 1
plausible-hosting-plausible_db-1         |
plausible-hosting-plausible_db-1         | PostgreSQL Database directory appears to contain a database; Skipping initialization
plausible-hosting-plausible_db-1         |
plausible-hosting-plausible_db-1         | 2023-03-26 12:47:21.203 UTC [1] FATAL:  database files are incompatible with server
plausible-hosting-plausible_db-1         | 2023-03-26 12:47:21.203 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 12, which is not compatible with this version 14.7
``` 

I've tried to update the old instance of Plausible (thinking that there might be some kind of migration going on) to a newer version and mounting them to the newer instance again, but the error persists. 

I don't want to run an older version of Plausible. Is there a way to fix this somehow else?

**Comment by ruslandoga** [comment, marked as answer]
👋 @chamik 

I think in order to restore your data in v14 you'd have to use `pg_dump` + `pg_restore` (or `psql` if using plain SQL for backup) instead of just archiving the data directory. I think the reason is, different major versions might have different data layout.

Here's an example guide: https://github.com/plausible/hosting/blob/master/upgrade/postgres.md

**Reply by chamik** [reply]
Worked like a charm. Thanks!
